flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/data/writer/AbstractBinaryWriter.java

```diff
@@ -165,8 +165,11 @@ public void writeDecimal(int pos, DecimalData value, int precision) {
         assert value == null || (value.precision() == precision);
 
         if (DecimalData.isCompact(precision)) {
-            assert value != null;
-            writeLong(pos, value.toUnscaledLong());
+            if (value == null) {
+                setNullBit(pos);
```

Comment:
why is this `setNullBit(pos); `but all the others are  `setNullAt(pos);`

Reply:
Good catch, I reviewed the code, `setNullBit` only sets the null-tracking bit of this field, and `setNullAt` sets null-bit and sets the fixed-length data to 0, so it should be `setNullAt(pos)`.